dashboard/app/controllers/programming_expressions_controller.rb

```diff
@@ -44,8 +44,12 @@ def update
       render :not_found
       return
     end
-    programming_expression.assign_attributes(programming_expression_params.except(:parameters))
+    programming_expression.assign_attributes(programming_expression_params.except(:category_key, :parameters))
     programming_expression.palette_params = programming_expression_params[:parameters]
+    programming_environment_category = programming_expression.programming_environment.categories.find_by_key(programming_expression_params[:category_key])
+    programming_expression.programming_environment_category_id = programming_environment_category&.id
+    # TODO: get rid of this when we remove the category column
```

Comment:
Is this the issue with search on lesson  edit you called out?

Reply:
Yes, it's related. The category column doesn't serve a function anymore so I was going to get rid of that column, but the lesson edit page allows searching by category.